zerver/views/users.py

```diff
@@ -123,13 +123,13 @@ def reactivate_user_backend(request: HttpRequest, user_profile: UserProfile,
     do_reactivate_user(target, acting_user=user_profile)
     return json_success()
 
-check_profile_data: Validator[List[Dict[str, Optional[Union[int, str, List[int]]]]]] = check_list(
-    check_dict(
-        [('id', check_int)],
-        value_validator=check_none_or(check_union([
-            check_int, check_string, check_list(check_int),
-        ])),
-    ),
+check_profile_data: Validator[List[Dict[str, Any]]] = check_list(
+    check_dict([
```

Comment:
The error you got here was correct: you forgot to change this `check_dict` to `check_dict_only`.